### PR TITLE
Implement `dpnp.logaddexp2` function

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR adds implementation of `dpnp.logaddexp2` function.
New binary universal function `_logaddexp2` as added to `_ufunc_impl` pybind11 extension.

The all affected tests are updated to cover different use cases.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_trigonometric.py [hunk 1]
```diff
@@ -1412,6 +1413,83 @@ def cumlogsumexp(
...
+
+    Calculates ``log2(2**x1 + 2**x2)``. This function is useful in machine
+    learning when the calculated probabilities of events may be so small as
+    to exceed the range of normal floating point numbers.  In such cases
+    the base-2 logarithm of the calculated probability can be used instead.
+    This function allows adding probabilities stored in such a fashion.
+
+This function calculates `log2(2**x1 + 2**x2)`. It is useful in machine
+learning when the calculated probabilities of events may be so small as
+to exceed the range of normal floating point numbers.  In such cases the base-2
```
Comment (vtavana) [inline dpnp/dpnp_iface_trigonometric.py:1431]:
```suggestion
to exceed the range of normal floating point numbers. In such cases the base-2
```

File: dpnp/dpnp_iface_trigonometric.py [hunk 2]
```diff
@@ -1412,6 +1413,83 @@ def cumlogsumexp(
...
+of the input array `x1` with the respective element `x2_i` of the input
+array `x2`.
+
+    Logarithm of the sum of exponentiations of the inputs in base-2.
+
+    Calculates ``log2(2**x1 + 2**x2)``. This function is useful in machine
+    learning when the calculated probabilities of events may be so small as
+    to exceed the range of normal floating point numbers.  In such cases
+    the base-2 logarithm of the calculated probability can be used instead.
+    This function allows adding probabilities stored in such a fashion.
```
Comment (vtavana) [inline dpnp/dpnp_iface_trigonometric.py:1427]: Do we need the indent at the beginning of each line here?

File: dpnp/dpnp_iface_trigonometric.py [hunk 3]
```diff
@@ -1412,6 +1413,83 @@ def cumlogsumexp(
...
+
+Limitations
+-----------
+Parameters `where` and `subok` are supported with their default values.
+Keyword arguments `kwargs` are currently unsupported.
+Otherwise ``NotImplementedError`` exception will be raised.
+
+See Also
+--------
+:obj:`dpnp.log` : Natural logarithm, element-wise.
```
Comment (vtavana) [inline dpnp/dpnp_iface_trigonometric.py:1469]: add `logaddexp` here and vice verse (add `logaddexp2` for See Also section of `_LOGADDEXP_DOCSTRING`.

File: dpnp/dpnp_iface_trigonometric.py [hunk 4]
```diff
@@ -1412,6 +1413,83 @@ def cumlogsumexp(
 )
 
 
+_LOGADDEXP2_DOCSTRING = """
+Calculates the logarithm of the sum of exponents for each element `x1_i`
```
Comment (vtavana) [inline dpnp/dpnp_iface_trigonometric.py:1417]:
```suggestion
Calculates the logarithm of the sum of exponents in base-2 for each element `x1_i`
```

File: dpnp/dpnp_iface_trigonometric.py [hunk 5]
```diff
@@ -1412,6 +1413,83 @@ def cumlogsumexp(
...
+of the input array `x1` with the respective element `x2_i` of the input
+array `x2`.
```
Comment (vtavana) [inline dpnp/dpnp_iface_trigonometric.py:1419]: Do we need this part at all? the following text already explains it.

File: dpnp/dpnp_iface_trigonometric.py (hunk 2 above)
Comment (antonwolfy) [reply dpnp/dpnp_iface_trigonometric.py:1427]: No, I will remove that, Thank you for noticing that.

File: dpnp/dpnp_iface_trigonometric.py (hunk 5 above)
Comment (antonwolfy) [reply dpnp/dpnp_iface_trigonometric.py:1419]: I'm going to remove part of the text below. Or do you mean at line#1429?

**Review by vtavana — APPROVED** [review]
Thank you, @antonwolfy!

**Pull request merged**